### DISCUSSION TITLE
Update is not returning the updated object

### DISCUSSION THREAD

**Discussion by jonathaneemmett** [supabase/supabase#13889] [Questions] [answered]
When I do an update of an object it does not return the updated object as this implies: https://supabase.com/docs/reference/javascript/v0/update

The payload I'm sending gets there and is correct and the return is data [], status is 200, statusText is ok, and err is null.. I have looked at this until I am cross eyed, so I'm reaching out.  

After some more digging it's not updating.  I can run a select and pull the record, so I know it's there.   Policy to update looks right + like I said no errors.   I even deleted policies and reset them just to make sure. 
    
    const { id, user_id, title, body, tags } = await request.json();
	console.log('PUT: ', id, user_id, title, body, tags);

       This works:

	const { data: draft, error } = await supabase.from('drafts').select('*').eq('id', id);
	console.log('Draft: ', draft, error);

      This does not, but return is [], 200, ok, null

	const {
		data,
		status,
		statusText,
		error: err
	} = await supabase
		.from('drafts')
		.update({
			user_id,
			title,
			body,
			tags,
			updated_at: new Date()
		})
		.eq('id', id)
		.select('*');

	console.log('Return: ', data, status, statusText, err);

**Comment by jonathaneemmett** [comment, marked as answer]
This ended up being a problem with the policy.  

    create policy "users can update their own drafts" on drafts for
    update
        using (auth.uid() = user_id)
        with check (auth.uid() = user_id);

**Reply by jonathassardinha** [reply]
Had the same issue, this saved me 😄

**Reply by MadeByDouglas** [reply]
same!

**Comment by sagar-jani** [comment]
I'm facing the same issue, it doesn't return anything, how did you change the policy to return something with update ?

**Comment by aaditya25052002** [comment]
what if I disable the RLs, it is still not working

**Comment by step135** [comment]
There should be information if data were changed or not. I reported this issue too!

**Reply by AbdullahAhmadAAK** [reply]
Hi @step135 
I tried the fix which @jonathaneemmett told above, and it worked.
Then I saw that I hadn't added the `.select()` at the end of my supabase query.You might wanna try that.

Removing the policy which Jonathan mentioned still has my functionality working fine btw.
